packages/commerce-sdk-react/README.md

```diff
@@ -64,7 +101,259 @@ const App = ({children}) => {
+```
+
+_💡 Hint: install and include `@tanstack/react-query-devtools` in your React app to see the queries (and their cache keys)._
+
+## Mutations
+
+The query hooks correspond to the http POST, PUT, PATCH, DELETE endpoints from the SCAPI API domains. The mutation hooks follow the signature pattern:
+
+```
+use<ApiName>Mutations(EndpointName)
+```
+
+For example, the [ShopperBaskets API](https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=Summary) has a number of endpoints, one of them being the [addItemToBasket](https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=addItemToBasket) endpoint (`POST /baskets/{basketId}/items`).
+
+```jsx
+import {useShopperBasketsMutation} from '@salesforce/commerce-sdk-react'
+
+const Example = ({basketId}) => {
+    // Typescript IDE intellisense for available options as string
+    const addItemToBasket = useShopperBasketsMutation('addItemToBasket')
+
+    return <button onClick={() => addItemToBasket.mutate({
+        parameters: {
+            basketId
+        },
+        body: {
+            productId: '25592770M',
+            price: 55,
+            quantity: 1
+        }
+    })} />
+}
+```
+
+You could also import the mutation options as a constant like:
+
+```jsx
+import {useShopperBasketsMutation, ShopperBasketsMutations} from '@salesforce/commerce-sdk-react'
+
+const Example = ({basketId}) => {
+    const addItemToBasket = useShopperBasketsMutation(ShopperBasketsMutations.AddItemToBasket)
+    return ...
+}
+```
+
+### Auto Cache Invalidations
+
+Since mutations changes data on the server, the cache entries that are potentially affected by the mutation is automatically invalidated.
+
+For example, a `addItemToBasket` mutation will automatically invalidate `getBasket`  and `customerBaskets` query
+
+## Ultilities
+
+Besides the collection of query hooks and mutation hooks, here are some ultility hooks to help you interact with SCAPI.
+
+### `useCommerceApi()`
+
+This hook returns a set of SCAPI clients, which are already initialized using the configurations passed to the provider. Note: this hook doesn't automatically include auth headers.
+
+```jsx
+import {useCommerceApi, useAccessToken} from '@salesforce/commerce-sdk-react'
+
+const Example = () => {
+    const api = useCommerceApi()
+    const {getTokenWhenReady} = useAccessToken()
+
+    const fetchProducts = async () => {
+        const token = await getTokenWhenReady()
+        const products = await api.shopperProducts.getProducts({
+                parameters: {ids: ids.join(',')},
+                headers: {
+                    Authorization: `Bearer ${token}`
+                }
+            })
+        return products
+    }
+
+}
+```
+
+### `useAccessToken()`
+
+```ts
+useAccessToken() => {token: String, getTokenWhenReady: Promise}
+```
+
+This ultility hook give access to the managed SLAS access token.
+
+### `useCustomerId()`
```

Comment:
Are we also planning on publishing and liking the typescript auto-generated docs site in the readme.md file with the rest of the utils and functions?